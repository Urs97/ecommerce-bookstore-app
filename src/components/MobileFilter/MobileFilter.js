import React, { useContext } from 'react';
import './MobileFilter.scss';

import StoreContext from '../../context/StoreContext';
import BookGenres from '../BookGenres/BookGenres';
import FilterByPrice from '../FilterByPrice/FilterByPrice';

const MobileFilter = () => {
    const context = useContext(StoreContext);
    const isOpen = context.state.mobileFilterPopup;

    return (
        <section className={`mobile-filter ${isOpen ? `` : `mobile-filter-closed`}`}>
            <button 
                className="button mobile-filter-close" 
                onClick={() => context.toggleMobileFilter('close')}
            >Close</button>
            <BookGenres />
            <FilterByPrice />
        </section>
    )
}

export default MobileFilter;