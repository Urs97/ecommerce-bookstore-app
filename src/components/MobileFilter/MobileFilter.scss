@import "../../styles/shared.scss";

.mobile-filter {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 2;
  background-color: #ffffff;
  width: 100%;
  height: min-content;
  font-size: 1.5em;
  padding: 2em 3em;
  padding-bottom: 3em;
  border-bottom: 0.25em solid $dark-blue;
  display: grid;
  grid-template-areas:
    "close-btn close-btn"
    "book-genres filter-by-price";
  grid-template-columns: 1fr 1fr;
  grid-template-rows: repeat(2, max-content);

  .mobile-filter-close {
    background-color: $dark-blue;
    margin-bottom: 1em;
    grid-area: close-btn;
    justify-self: center;
    align-self: center;
  }

  .mobile-filter-close:hover {
    background-color: $yellow;
  }

  .book-genres {
    width: 15em;
    text-align: center;
    grid-area: book-genres;
    justify-self: flex-start;
    margin-top: 2em;

    h4 {
      margin-bottom: 0.5em;
    }

    ul {
      list-style-type: none;
    }
  }

  .filter-by-price {
    width: 15em;
    grid-area: filter-by-price;
    justify-self: center;
    padding: 0;
    margin-top: 2em;
    border-bottom: none;

    h4 {
      text-align: center;
    }
  }

  .noUi-horizontal {
    margin-bottom: 2em;
  }
}

@media (max-width: 650px) {
  .mobile-filter {
    font-size: 1.25em;
  }
}

@media (max-width: 500px) {
  .mobile-filter {
    padding: 2em;
  }
}

@media (max-width: 460px) {
  .mobile-filter {
    font-size: 1.2em;
    display: flex;
    flex-direction: column;
    align-items: center;

    .book-genres {
      padding-bottom: 1em;
      border-bottom: 2px solid #f4f7f9;
    }

    .filter-by-price {
      margin-top: 1em;
      margin-right: 0;
    }
  }
}

@media (max-width: 370px) {
  .mobile-filter {
    font-size: 1.35em;
  }
}
@media (max-width: 340px) {
  .mobile-filter {
    font-size: 1.45em;
  }
}

.mobile-filter-closed {
  display: none;
}
