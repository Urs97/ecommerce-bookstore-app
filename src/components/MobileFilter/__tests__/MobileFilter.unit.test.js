import React from 'react';
import {shallow} from 'enzyme';
import MobileFilter from '../MobileFilter';

describe('MobileFilter', () => {
    test('renders MobileFilter', () => {
        const wrapper = shallow(
            <MobileFilter />
        );
        expect(wrapper).toMatchSnapshot();
    });
});