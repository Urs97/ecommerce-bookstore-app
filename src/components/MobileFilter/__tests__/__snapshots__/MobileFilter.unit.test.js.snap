// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileFilter renders MobileFilter 1`] = `
<section
  className="mobile-filter mobile-filter-closed"
>
  <button
    className="button mobile-filter-close"
    onClick={[Function]}
  >
    Close
  </button>
  <BookGenres />
  <FilterByPrice />
</section>
`;

exports[`MobileFilter renders empty MobileFilter 1`] = `
<section
  className="mobile-filter mobile-filter-closed"
>
  <button
    className="button mobile-filter-close"
    onClick={[Function]}
  >
    Close
  </button>
  <BookGenres />
  <FilterByPrice />
</section>
`;
